### ISSUE TITLE
Unit test errors with PHP 7.4

### ISSUE THREAD

**Issue by phil-davis** [owncloud/encryption#193]
https://drone.owncloud.com/owncloud/encryption/1233/10/4
```
There were 4 errors:

1) OCA\Encryption\Tests\Hooks\UserHooksTest::testSetPassphrase
Trying to access array offset on value of type null

/var/www/owncloud/server/apps/encryption/lib/Hooks/UserHooks.php:290
/var/www/owncloud/server/apps/encryption/tests/unit/Hooks/UserHooksTest.php:238

2) OCA\Encryption\Tests\Hooks\UserHooksTest::testSetPassphraseWithoutSession
Trying to access array offset on value of type null

/var/www/owncloud/server/apps/encryption/lib/Hooks/UserHooks.php:290
/var/www/owncloud/server/apps/encryption/tests/unit/Hooks/UserHooksTest.php:306

3) OCA\Encryption\Tests\Hooks\UserHooksTest::testSetPassphraseWithoutSessionLoggerError
Trying to access array offset on value of type null

/var/www/owncloud/server/apps/encryption/lib/Hooks/UserHooks.php:290
/var/www/owncloud/server/apps/encryption/tests/unit/Hooks/UserHooksTest.php:342

4) OCA\Encryption\Tests\Hooks\UserHooksTest::testSetPasswordNoUser
Trying to access array offset on value of type null

/var/www/owncloud/server/apps/encryption/lib/Hooks/UserHooks.php:290
/var/www/owncloud/server/apps/encryption/tests/unit/Hooks/UserHooksTest.php:378

ERRORS!
Tests: 251, Assertions: 707, Errors: 4.
```

**Comment by phil-davis** [comment]
https://github.com/owncloud/encryption/blob/master/tests/unit/Hooks/UserHooksTest.php#L189
```
	public function testSetPassphrase() {
		$this->sessionMock->expects($this->exactly(4))
			->method('getPrivateKey')
			->willReturnOnConsecutiveCalls(true, false);
...
```
It gets called 4 times (when `setPassphrase` is called each of 4 times in the test), that is fine. But why are only 2 values specified in `willReturnOnConsecutiveCalls`

And `testSetPassphrase` does a whole lot of stuff setting up to supposedly `// Test first if statement` and `// Test Second if conditional` and `// Test third and final if condition` - it is not so obvious exactly which code blocks in `UserHooks.php` `setPassphrase` are intended to be executed at each point, and which ones are actually getting executed.

I am suspicious that at some points in the test, the intended "code-under-test" is not the code path that actually happens, and so `createKeyPair` is getting called in a place where it is not ecpected and returns null...

Smeone needs to know or understand exactly what the intended flow of this test is, so all the various mocks and expects can be sorted out.

**Comment by phil-davis** [comment]
Code that fixes this was in PR #197 and was cherry-picked into PR #192 
That is all passing and merged.
